inference-engine/src/mkldnn_plugin/mkldnn_weights_cache.cpp

```diff
@@ -25,11 +25,11 @@ MKLDNNWeightsSharing::MKLDNNSharedMemory::operator MKLDNNMemoryPtr() const {
 }
 
 bool MKLDNNWeightsSharing::MKLDNNSharedMemory::isValid() const {
-    return memory->valid;
+    return memory->valid.load(std::memory_order_acquire);
 }
 
 void MKLDNNWeightsSharing::MKLDNNSharedMemory::valid(bool b) {
-    memory->valid = b;
+    memory->valid.store(b, std::memory_order_release);
```

Comment:
Do we really need any memory access synchronization other than the `valid` flag itself? If not, it seems that relaxed ordering should be enough.

Reply:
Discussed offline. We need such ordering indeed.

Reply:
All graphNode->execute(...) operations sholud be completed before MKLDNNSharedMemory::valid. So I think it is better to use memory_order_release.